common/scala/src/main/scala/org/apache/openwhisk/core/entity/WhiskEntity.scala

```diff
@@ -115,15 +116,25 @@ object WhiskEntity {
 
 object WhiskDocumentReader extends DocumentReader {
   override def read[A](ma: Manifest[A], value: JsValue) = {
-    ma.runtimeClass match {
-      case x if x == classOf[WhiskAction]         => WhiskAction.serdes.read(value)
+    val doc = ma.runtimeClass match {
+      case x if x == classOf[WhiskAction] => WhiskAction.serdes.read(value)
       case x if x == classOf[WhiskActionMetaData] => WhiskActionMetaData.serdes.read(value)
-      case x if x == classOf[WhiskPackage]        => WhiskPackage.serdes.read(value)
-      case x if x == classOf[WhiskActivation]     => WhiskActivation.serdes.read(value)
-      case x if x == classOf[WhiskTrigger]        => WhiskTrigger.serdes.read(value)
-      case x if x == classOf[WhiskRule]           => WhiskRule.serdes.read(value)
-      case _                                      => throw DocumentUnreadable(Messages.corruptedEntity)
+      case x if x == classOf[WhiskPackage] => WhiskPackage.serdes.read(value)
+      case x if x == classOf[WhiskActivation] => WhiskActivation.serdes.read(value)
+      case x if x == classOf[WhiskTrigger] => WhiskTrigger.serdes.read(value)
+      case x if x == classOf[WhiskRule] => WhiskRule.serdes.read(value)
+      case _ => throw DocumentUnreadable(Messages.corruptedEntity)
     }
+    value.asJsObject.fields.get("entityType") match {
+      case Some(entityType) => {
```

Comment:
This can be simplified a bit via destructuring the `JsString` itself and avoid explicit conversions later

```patch
- case Some(entityType) => {
+ case Some(JsString(entityType)) => {
```